ui/src/views/network/NicsTab.vue

```diff
@@ -368,9 +376,26 @@ export default {
         this.$pollJob({
           jobId: response.addnictovirtualmachineresponse.jobid,
           successMessage: this.$t('message.success.add.network'),
-          successMethod: () => {
-            this.loadingNic = false
-            this.closeModals()
+          successMethod: async () => {
+            if (this.addNetworkData.makedefault) {
+              try {
+                this.nic = await this.getNic(params.networkid, params.virtualmachineid)
+                if (this.nic) {
+                  this.setAsDefault(this.nic)
+                } else {
+                  this.$notifyError('NIC data not found.')
+                  this.loadingNic = false
+                  this.closeModals()
+                }
+              } catch (error) {
+                this.$notifyError('Failed to fetch NIC data.')
+                this.loadingNic = false
+                this.closeModals()
+              }
+            } else {
+              this.loadingNic = false
```

Comment:
can the following be moved to the line before `errorMessage` ?

```
              this.loadingNic = false
              this.closeModals()
```

Reply:
Made some changes to the code. Before that the active tab was getting reset to Details on Add-Network.
Now the Network tab remains the active tab after adding a network.
Please check if this comment still applies.

Reply:
@abh1sar 
the changes in your last commit https://github.com/apache/cloudstack/pull/9708/commits/02be6a2e48e906d5256ab683747d62709bb73a52 is exactly what I meant.